test/network/network.test.ts

```diff
@@ -0,0 +1,241 @@
+import { HttpClient } from '../../src/shared/http/client';
+import sinon from 'sinon';
+import { AppDetails } from '../../src/shared';
+import { Network } from '../../src/network/index';
+import { headersWithBasicAuth } from '../../src/shared/headers/index';
+import {
+  StartUploadPayload,
+  FinishUploadPayload,
+  FinishUploadResponse,
+  StartUploadResponse,
+} from '../../src/network/types';
+
+const httpClient = HttpClient.create('');
+
+const validUUID = 'b541b138-a6f2-4729-8d20-8e6011cb8216';
+const validHex = '2e1884c34f174110ca6e324e7b745754b3d6356b53ef9f594b960fd534050089';
+
+describe('network ', () => {
+  beforeEach(() => {
+    sinon.stub(HttpClient, 'create').returns(httpClient);
+  });
+
+  afterEach(() => {
+    sinon.restore();
+  });
+
+  describe('validate inputs', () => {
+    it('Validates startUpload', async () => {
+      // Arrange
+      const { client } = clientAndHeadersWithBasicAuth();
+      const idBucket = 'id-bucket';
+      const invalidIndexPayload: StartUploadPayload = {
+        uploads: [{ index: 'invalid-hex', size: 40 }],
+      };
+
+      // Act
+      try {
+        const promise = await client.startUpload(idBucket, invalidIndexPayload);
+        expect(promise).toBeUndefined();
+      } catch (err) {
+        // Assert
+        expect(err).toBeDefined();
+        expect(err).toEqual(new Error('Invalid index'));
+      }
+
+      // Arrange
+      const invalidSizePayload: StartUploadPayload = {
+        uploads: [{ index: validHex, size: -33 }],
+      };
+
+      // Act
+      try {
+        const promise = await client.startUpload(idBucket, invalidSizePayload);
+        expect(promise).toBeUndefined();
+      } catch (err) {
+        // Assert
+        expect(err).toBeDefined();
```

Comment:
This check could be removed as the next line will not pass the test if `err` is not defined